provider.go

```diff
@@ -1,47 +1,182 @@
-// Package libdnstemplate implements a DNS record management client compatible
-// with the libdns interfaces for <PROVIDER NAME>. TODO: This package is a
-// template only. Customize all godocs for actual implementation.
-package libdnstemplate
+package autodns
 
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"slices"
 
 	"github.com/libdns/libdns"
 )
 
-// TODO: Providers must not require additional provisioning steps by the callers; it
-// should work simply by populating a struct and calling methods on it. If your DNS
-// service requires long-lived state or some extra provisioning step, do it implicitly
-// when methods are called; sync.Once can help with this, and/or you can use a
-// sync.(RW)Mutex in your Provider struct to synchronize implicit provisioning.
+const (
+	autoDNSendpoint string = "https://api.autodns.com/v1"
+	autoDNScontext  string = "4"
+)
 
-// Provider facilitates DNS record manipulation with <TODO: PROVIDER NAME>.
+// Provider facilitates DNS record manipulation with Autodns.
 type Provider struct {
-	// TODO: put config fields here (with snake_case json
-	// struct tags on exported fields), for example:
-	APIToken string `json:"api_token,omitempty"`
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	Endpoint   string `json:"Endpoint"`
+	Context    string `json:"context"`
+	Primary    string `json:"primary"`
+	httpClient *http.Client
+}
+
+// NewWithDefaults is a convenience method to create the provider with sensible defaults.
+func NewWithDefaults(username, password string) *Provider {
```

Comment:
We can't have constructor methods, since they are not part of the libdns interfaces (users would have to write separate code for each provider if they used constructors, kind of defeating the purpose).

The type will need to be usable without a constructor. See https://github.com/libdns/libdns/wiki/Implementing-a-libdns-package#requirements

Reply:
@mholt it's completely usable without it, I just added it for convenience. :) I can add a few guards though to ensure everything is actually set. Would that work?

Reply:
Resolve :)

Reply:
Great!